webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/EditTableRow.js

```diff
@@ -0,0 +1,211 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Tr, Td } from '@patternfly/react-table';
+import { TimesIcon, CheckIcon } from '@patternfly/react-icons';
+import {
+  SelectVariant,
+  Select,
+  SelectOption,
+  Button,
+  TextInput,
+  TextArea,
+  Checkbox,
+  FormGroup,
+  Form,
+  Tooltip,
+} from '@patternfly/react-core';
+import { APIActions } from '../../../../redux/API';
+import { sprintf, translate as __ } from '../../../../common/I18n';
+import { HOST_PARAM, columnNames, typeOptions } from './ParametersConstants';
+import { updateHost } from '../../ActionsBar/actions';
+import { RowActions } from './RowActions';
+
+export const EditParametersTableRow = ({
+  param,
+  rowIndex,
+  setEditingRow,
+  hostId,
+  editHostsPermission,
+  isNew,
+}) => {
+  const [name, setName] = useState(param.name);
+  const [type, setType] = useState(param.parameter_type);
+  const [value, setValue] = useState(param.value);
+  const [selectIsOpen, setSelectIsOpen] = useState(false);
+  const [selectValueIsOpen, setSelectValueIsOpen] = useState(false);
+  const [isHide, setIsHide] = useState(param['hidden_value?']);
+
+  const dispatch = useDispatch();
+
+  const APIActionsProps = {
+    errorToast: ({ response }) =>
+      // eslint-disable-next-line camelcase
+      response?.data?.error?.full_messages || response?.data?.error?.message,
+    handleSuccess: () => {
+      dispatch(updateHost(hostId));
+    },
+  };
+  const onSubmit = () => {
+    if (isNew || param.associated_type !== HOST_PARAM) {
+      dispatch(
+        APIActions.post({
+          url: `/api/hosts/${hostId}/parameters/`,
+          key: 'NEW-PARAM',
+          params: {
+            name,
+            value,
+            parameter_type: type,
+            hidden_value: isHide,
+          },
+          successToast: () =>
+            sprintf(__('Parameter %s has been created successfully'), name),
+          ...APIActionsProps,
+        })
+      );
+    } else {
+      dispatch(
+        APIActions.put({
+          url: `/api/hosts/${hostId}/parameters/${param.id}`,
+          key: `${param.id}-EDIT`,
+          params: {
+            name,
+            value,
+            parameter_type: type,
+            hidden_value: isHide,
+          },
+          successToast: () =>
+            sprintf(__('Parameter %s has been edited successfully'), name),
+          ...APIActionsProps,
+        })
+      );
+    }
+  };
+  return (
+    <Tr key={rowIndex}>
+      <Td dataLabel={columnNames.name}>
+        <Form>
+          <FormGroup
+            validated={name.includes(' ') ? 'error' : null}
+            helperTextInvalid={__("Name can't contain spaces")}
+          >
+            <TextInput
+              validated={name.includes(' ') ? 'error' : null}
+              aria-label={`${param.name} name text`}
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label the `aria-label` is mainly for screen readers. I wonder why we use this instead of the regular `label`. Also, shouldn't `aria-label` be translated?

Reply:
I use aria-label as its part of pf4 examples. I'm not sure if they should be translated because there was some issue with tests when there are translated html attributes, like here: https://projects.theforeman.org/issues/35630. If they should be translated lets do it to all `aria-label`s in foreman in a different pr

Reply:
In Katello `aria-label` is also  useful for writing React Testing Library tests: https://testing-library.com/docs/queries/byrole will search for DOM elements by aria-label, among other things.

Reply:
Also I know qe uses them is some tests